mirdata/datasets/slakh.py

```diff
@@ -0,0 +1,408 @@
+    
+    The original release of Slakh, called Slakh2100, 
+    contains 2100 automatically mixed tracks and accompanying, aligned MIDI files, 
+    synthesized from 187 instrument patches categorized into 34 classes, totaling 
+    145 hours of mixture data.
+
+    This loader supports two versions of Slakh:
+    - Slakh2100-redux: a deduplicated version of Slack2100 containing 1710 multitracks
+    - baby-slakh: a mini version with 16k wav audio and only the first 20 tracks
+
+    This dataset was created at Mitsubishi Electric Research Labl (MERL) and
+    Interactive Audio Lab at Northwestern University by Ethan Manilow, 
+    Gordon Wichern, Prem Seetharaman, and Jonathan Le Roux.
+
+    For more information see http://www.slakh.com/
+
+"""
+import os
+from typing import BinaryIO, Optional, Tuple
+
+import librosa
+import numpy as np
+import pretty_midi
+import yaml
+
+from mirdata import io, download_utils, jams_utils, core, annotations
+
+BIBTEX = """
+@inproceedings{manilow2019cutting,
+  title={Cutting Music Source Separation Some {Slakh}: A Dataset to Study the Impact of Training Data Quality and Quantity},
+  author={Manilow, Ethan and Wichern, Gordon and Seetharaman, Prem and Le Roux, Jonathan},
+  booktitle={Proc. IEEE Workshop on Applications of Signal Processing to Audio and Acoustics (WASPAA)},
+  year={2019},
+  organization={IEEE}
+}
+"""
+
+INDEXES = {
+    "default": "2100-redux",
+    "test": "baby",
+    "2100-redux": core.Index(
+        filename="slakh_index_2100-redux.json", partial_download=["2100-redux"]
+    ),
+    "baby": core.Index(filename="slakh_index_baby.json", partial_download=["baby"]),
+}
+
+REMOTES = {
+    "2100-redux": download_utils.RemoteFileMetadata(
+        filename="slakh2100_flac_redux.tar.gz",
+        url="https://zenodo.org/record/4599666/files/slakh2100_flac_redux.tar.gz?download=1",
+        checksum="f4b71b6c45ac9b506f59788456b3f0c4",
+    ),
+    "baby": download_utils.RemoteFileMetadata(
+        filename="babyslakh_16k.tar.gz",
+        url="https://zenodo.org/record/4603870/files/babyslakh_16k.tar.gz?download=1",
+        checksum="311096dc2bde7d61c97e930edbfc7f78",
+    ),
+}
+
+LICENSE_INFO = """
+Creative Commons Attribution 4.0 International
+"""
+
+SPLITS = ["train", "validation", "test", "omitted"]
+
+#: Mixing group to program number mapping
+MIXING_GROUPS = {
+    "piano": [0, 1, 2, 3, 4, 5, 6, 7],
+    "guitar": [24, 25, 26, 27, 28, 29, 30, 31],
+    "bass": [32, 33, 34, 35, 36, 37, 38, 39],
+    "drums": [128],
+}
+
+
+class Track(core.Track):
+    """slakh Track class
+
+    Attributes:
+        audio_path (str or None): path to the track's audio file. For some unusual tracks,
+            such as sound effects, there is no audio and this attribute is None.
+        data_split (str or None): one of 'train', 'validation', 'test', or 'omitted'.
+            'omitted' tracks are part of slack2100-redux which were found to be
+            duplicates in the original slackh2011.
```

Comment:
Typo: `original slackh2011` --> `original Slakh2100`. FWIW, it's up to you if you want to capitalize "slakh" or not. I don't think we decided on stylizing it one way or another.